frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkerThread.java

```diff
@@ -292,9 +294,28 @@ void retry() {
             return;
         }
 
-        logger.info("Retry worker: {}", workerId);
-        ModelManager modelManager = ModelManager.getInstance();
-        modelManager.submitTask(this);
+        running.set(false);
+        ModelManager manager = ModelManager.getInstance();
+
+        backoffIdx = (backoffIdx + 1) < Model.BACKOFF.length ? backoffIdx + 1 : backoffIdx;
+
+        Runnable work =
+                () -> {
+                    try {
+                        logger.info("Waiting for shutdown latch");
+                        // Send poison as control command
+                        model.addJob(
+                                workerId,
+                                new Job(null, model.getModelName(), WorkerCommands.STOP, null));
+                        shutdownLatch.await();
+                        manager.submitTask(this);
+                    } catch (InterruptedException e) {
+                        logger.error("", e);
+                    }
+                };
+
+        manager.getScheduler().schedule(work, Model.BACKOFF[backoffIdx], TimeUnit.MINUTES);
```

Comment:
Backoff by minutes seems too large for initial couple of retry.

Reply:
I thought we agreed on this backoff mechanism. I can add a couple of options for seconds as well.